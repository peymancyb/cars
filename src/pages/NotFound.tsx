import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Link from '@material-ui/core/Link';
import Layout from '../components/Layout';

import Logo from '../assets/logo@2x.png';

const useStyles = makeStyles(() => ({
  container: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    flexDirection: 'column',
    height: '80vh',
  },
  link: {
    color: '#dd843e',
    padding: 4,
  },
  logo: {
    width: 300,
  },
}));


function NotFound() {
  const classes = useStyles();

  return (
    <Layout>
      <div className={classes.container}>
        <img src={Logo} className={classes.logo} alt="logo" />
        <h1>404 - Not Found</h1>
        <p>Sorry, the page you are looking for does not exist.</p>
        <p>
          You can always go back to
          <span>
            <Link
              href="/"
              className={classes.link}
            >
              homepage.
            </Link>
          </span>
        </p>
      </div>
    </Layout>
  );
}

export default NotFound;
