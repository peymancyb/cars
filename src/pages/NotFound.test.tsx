import React from 'react';
import {render, cleanup, fireEvent} from '@testing-library/react';
import {createBrowserHistory} from 'history';
import NotFound from './NotFound';

afterEach(cleanup);

test('(NotFound) component should match snapshot', () => {
  const {asFragment} = render(<NotFound />);
  expect(asFragment).toMatchSnapshot();
});

test('(NotFound) component renders correctly', () => {
  const {getByTestId} = render(<NotFound />);
  expect(getByTestId('not-found-component')).toHaveTextContent(
    '404 - Not Found',
  );
});

test('By clicking on "homepage" link it should navigate to home page', () => {
  const history = createBrowserHistory();
  const {getByText} = render(<NotFound />);

  fireEvent.click(getByText('homepage.'));
  expect(history.location.pathname).toBe('/');
});
