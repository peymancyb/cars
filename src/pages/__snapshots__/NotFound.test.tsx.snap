// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(NotFound) component should match snapshot 1`] = `[Function]`;
